base/libc.jl

```diff
@@ -400,4 +403,27 @@ Interface to the C `srand(seed)` function.
 """
 srand(seed=floor(Int, time()) % Cuint) = ccall(:srand, Cvoid, (Cuint,), seed)
 
+"""
+    mkfifo(path::AbstractString, [mode::Integer]) -> path
+
+Make a FIFO special file (a named pipe) at `path`.  Return `path` as-is on success.
+
+!!! compat "Julia 1.5"
+    `mkfifo` requires at least Julia 1.5.
+"""
+function mkfifo end
+
+if Sys.isunix()
```

Comment:
Maybe better to move this check inside the method and give a clearer "not supported on this platform" error instead of a MethodError?

Reply:
I actually thought it's a nice pattern to not provide the method at all so that you can use `hasmethod` etc. to check if some API is implemented in a given platform. I thought it's nicer than each caller to encode such information in their code.

Reply:
That's an interesting point, but I don't think `hasmethod` can be used consistently to check for platform support. For example, there could be cases where certain argument values aren't supported on a certain OS.

Reply:
OK. It makes sense.